src/main/kotlin/javabot/dao/impl/WeatherDao.kt

```diff
@@ -1,56 +1,24 @@
 package javabot.dao.impl
 
+import com.google.inject.Inject
+import javabot.JavabotConfig
 import java.util.Arrays
-import java.util.HashMap
-
-import javabot.dao.WeatherHandler
+import javabot.dao.impl.openweathermap.OpenWeatherMapHandler
 import javabot.model.Weather
 
-/**
- * Implements a weather service Dao using Weather Underground's Weather service, if that fails it tries Google's.
- *
- *
- * The user can also specify "using google for [place]" or "using wu for [place]" if they want to retrieve weather data
- * from a specific API
-
- */
-class WeatherDao {
-
+class WeatherDao @Inject constructor(private val javabotConfig: JavabotConfig) {
     //add more handlers to the list if there are other weather sources you want
     //the operation to support
     private val handlers = Arrays.asList(
```

Comment:
with only one handler, this list seems overkill.  we could just @Inject the instance then inject `javabotConfig` reference automatically...

Reply:
Well, the *design* is such that there was a cascading feature: theoretically you could have a weatherunderground api, an openweather api, a yahoo weather api... and it would iterate through until it got a valid response out of the service. It's overkill but mostly because WU killed their service.